libraries/AP_HAL_Linux/OpticalFlow_Onboard.cpp

```diff
@@ -90,18 +96,37 @@ void OpticalFlow_Onboard::init(AP_HAL::OpticalFlow::Gyro_Cb get_gyro)
                       "video device");
     }
 
-    if (!_videoin->set_crop(left, top, crop_width, crop_height)) {
-        AP_HAL::panic("OpticalFlow_Onboard: couldn't set video crop");
-    }
-
     if (!_videoin->set_format(&_width, &_height, &_format, &_bytesperline,
                               &_sizeimage)) {
         AP_HAL::panic("OpticalFlow_Onboard: couldn't set video format");
         return;
     }
 
-    if (_format != V4L2_PIX_FMT_NV12 && _format != V4L2_PIX_FMT_GREY) {
-        AP_HAL::panic("OpticalFlow_Onboard: planar or monochrome format needed");
+    if (_format != V4L2_PIX_FMT_NV12 && _format != V4L2_PIX_FMT_GREY && _format != V4L2_PIX_FMT_YUYV) {
+        AP_HAL::panic("OpticalFlow_Onboard: format not supported\n");
+    }
+
+    if (!_videoin->get_capabilities_flags(&cap_flags)) {
+        AP_HAL::panic("OpticalFlow_Onboard: couldn't get capabilities flags");
+    }
+
+    if (cap_flags & V4L2_CAP_VIDEO_OVERLAY) {
```

Comment:
This is not the correct check to see if crop is supported http://linuxtv.org/downloads/v4l-dvb-apis/overlay.html
Overlay cap is something related to overlaying video on an output interface.
You can check if the S_CROP ioctl returns an error and the errno set to ENOTTY to see if crop is supported.